docs/src/hugo/content/documentation/configuration.md

```diff
@@ -0,0 +1,691 @@
+
+```
+nelson.docker.verify-tls = true
+``` 
+
+----
+
+## Email
+
+Nelson can notify you by Email when changes to deployments happen. In order to do this, Nelson needs to be configured with an SMTP server. This is fully compatible with public cloud email offerings like SES (or any other provider that implements the SMTP protocol).
+
+* [email.host](#email-host)
+* [email.port](#email-port)
+* [email.from](#email-from)
+* [email.user](#email-user)
+* [email.password](#email-password)
+
+#### email.host
+
+Controls where Nelson will look for your SMTP email server. 
+
+```
+nelson.email.host = "mail.company.com"
+```
+
+#### email.port
+
+What port should Nelson use when talking to your SMTP email server.
+
+```
+nelson.email.port = 9000
+```
+
+#### email.from
+
+When Nelson sends emails about system status, what should the `From` line in the email be?
+
+```
+nelson.email.from = "nelson@example.com"
+```
+
+#### email.user
+
+If your SMTP server requires authentication, what username should be used.
+
+```
+nelson.email.user = "someuser"
+```
+
+#### email.password
+
+If your SMTP server requires authentication, what password should be used.
+
+```
+nelson.email.password = "somepassword"
+```
+
+----
+
+## Github
+
+Nelson requires a set of Github credentials in order to be able to interact with your source code. For more information, please [see the installation section](/getting-started/install.html#nelson-server).
+
+* [github.client-id](#github-client-id)
+* [github.client-secret](#github-client-secret)
+* [github.github.scope](#github-github.scope)
+* [github.access-token](#github-access-token)
+* [github.system-username](#github-system-username)
+
+
+#### github.client-id
+
+```
+nelson.github.client-id = "xxxxxxxxxxx"
+```
+
+#### github.client-secret
+
+```
+nelson.github.client-secret = "yyyyy"
+```
+
+#### github.redirect-uri
+
+What is the fully qualified URL that Github will redirect the OAuth exchange back too. This should match the domain configured in `network.external-host` and `network.external-port`, if configured.
+
+```
+nelson.github.redirect-uri = "https://nelson.local/auth/exchange"
+```
+
+#### github.scope
+
+What OAuth scopes should Nelson use when interacting with Github. Read more about Github OAuth scopes [on the Github documentation site](https://developer.github.com/apps/building-oauth-apps/understanding-scopes-for-oauth-apps/). 
+
+```
+nelson.github.scope  = "repo"
+```
+
+#### github.system-username
+
+Configure the Github username that Nelson will conduct operations as. This is needed because some operations that Nelson does are not in the context of any human interaction. For example, when Nelson conducts a deployment, it's all happening automatically and asynchronously from any human, and thus human-user OAuth tokens. To protect privacy Nelson does not store end-user OAuth tokens, and such needs its own user to interact with your Github account.  
```

Comment:
```suggestion
Configure the Github username that Nelson will conduct operations as. This is needed because some operations that Nelson does are not in the context of any human interaction. For example, when Nelson conducts a deployment, it's all happening automatically and asynchronously from any human, and thus human-user OAuth tokens cannot be utilized within these operations. To protect privacy Nelson does not store end-user OAuth tokens, and as such needs its own user to interact with your Github account.  
```